### ISSUE TITLE
Why rethrow `ConfigException` in `Mailer`?

### ISSUE THREAD

**Issue by mdawaffe** [flaviovs/vault#6]
https://github.com/flaviovs/vault/blob/5455673be8e1f7f19aa7bfe7eaac9b90a60f420f/lib/util.php#L47-L57

Could the message in `ConfigException` be expanded, or is it better to rethrow a new `VaultException`?

**Comment by flaviovs** [comment]
I think that you mean why **not** rethrow `ConfigException` in `Mailer`, right?

`ConfigException` is thrown by the config object whenever you try to access a nonexistent (section, key). In fact, the config object may throw `ConfigSectionNotFoundException` or `ConfigKeyNotFoundException` if is the section or the key that can not be found, respectively. Since both descend from `ConfigException`, and we don't care if it is the section or the key that is missing in this case -- we just want to ensure that the mailer is _always_ configured -- we try to catch the common ancestor exception.

Regarding rethrowing as `VaultException`: the config object had a different interface at the beginning of the project (it was a pure array) , and there was an _if_ test that used to throw `VaultException` if the values were empty. After I converted array access to a proper object method call, I thought that it made sense to keep the Mailer construct interface. There's a slight semantic difference between raising _Vault_ instead of _Config_ exception, but since this is done very early in during system init, and this is really an exceptional situation, it doesn't really matter much what exception is throw as long it prevents the system from continuing, so I decided to leave it as is.

Notice that there's a small advantage in not letting `ConfigException` bubble up: the admin will get a more precise error message from the exception, instead of something like `ConfigSectionNotFound: debug').
